server/EmeraldChameleonChat.Services/Hubs/ChatHub.cs

```diff
@@ -0,0 +1,59 @@
+﻿using EmeraldChameleonChat.Services.AutoMapperProfiles;
+using EmeraldChameleonChat.Services.DAL.DbContexts;
+using EmeraldChameleonChat.Services.DAL.Repository.RepositoryInterfaces;
+using EmeraldChameleonChat.Services.Model.Entity;
+using EmeraldChameleonChat.Services.Model.DTO;
+using Microsoft.AspNetCore.SignalR;
+using EmeraldChameleonChat.Services.DAL.Repository;
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
+using AutoMapper;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.Logging;
+
+namespace EmeraldChameleonChat.Hubs
+{
+    public class ChatHub : Hub
+    {
+        private readonly ILogger<ChatHub> _logger;
+        private readonly IChatRoomMessageRepository _context;
+        private readonly IMapper _mapper;
+        public ChatHub(ILogger<ChatHub> logger, IChatRoomMessageRepository context, IMapper mapper)
+        {
+            _logger = logger;
+            _context = context;
```

Comment:
Maybe rename _context to _chatRoomRepo? 

I thought you were using the DbContext

Reply:
I am using it. I can rename these so its more readable. Thanks.